dbt/adapters/athena/impl.py

```diff
@@ -57,6 +57,46 @@ def convert_number_type(cls, agate_table: agate.Table, col_idx: int) -> str:
     def convert_datetime_type(cls, agate_table: agate.Table, col_idx: int) -> str:
         return "timestamp"
 
+    # TODO: Add more lf-tag unit tests when moto supports lakeformation
+    # moto issue: https://github.com/getmoto/moto/issues/5964
+    @available
+    def add_lf_tags(self, database: str, table: str = None, lf_tags: Dict[str, str] = {}):
+        conn = self.connections.get_thread_connection()
+        client = conn.handle
+
+        if not lf_tags:
+            lf_tags = conn.credentials.lf_tags
```

Comment:
what if conn.credentials.lf_tags are empty? you might want to add another check on that.
```
if lf_tags:
  add tags..
else:
  logger.debug('skipping lakeformation tagging as empty...') 
```

Reply:
Good catch, added an explicit check let me know if you had something else in mind

Reply:
Just pushed an update that makes the logic slightly cleaner